File: navigation/src/main/java/ru/touchin/roboswag/components/navigation/activities/OnBackPressedListener.java
package ru.touchin.roboswag.components.navigation.activities;

/**
 * Interface to be implemented for someone who want to intercept device back button pressing event.
 */
public interface OnBackPressedListener {

    /**
     * Calls when user presses device back button.
     *
     * @return True if it is processed by this object.
     */
    boolean onBackPressed();

}

File: navigation/src/main/java/ru/touchin/roboswag/components/navigation/activities/OnBackPressedListener.kt
package ru.touchin.roboswag.components.navigation.activities

/**
 * Interface to be implemented for someone who want to intercept device back button pressing event.
 */
interface OnBackPressedListener {

    /**
     * Calls when user presses device back button.
     *
     * @return True if it is processed by this object.
     */
    fun onBackPressed(): Boolean

}
